.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,52 @@
+name: CodeQL Analysis
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+
+jobs:
+  analyze:
+    name: CodeQL Analysis
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        id: init_codeql
+        uses: github/codeql-action/init@v1
+        with:
+          queries: security-and-quality
+
+      - name: Build projects
+        id: build_projects
+        shell: pwsh
+        run: |
+          $projectsArray = @(
+            '.\src\Microsoft.OpenApi\Microsoft.OpenApi.csproj',
+            '.\src\Microsoft.OpenApi.Readers\Microsoft.OpenApi.Readers.csproj',
+            '.\src\Microsoft.OpenApi.Tool\Microsoft.OpenApi.Tool.csproj'
+          )
+          $targetsArray = @(
+            'restore',
+            'build'
+          )
+
+          $projectsArray | ForEach-Object {
+            $project = $PSItem
+            
+            $targetsArray | ForEach-Object {
+              dotnet msbuild $project `
```

Comment:
```suggestion
              dotnet build $project `
```

Reply:
We took direction for the `dotnet msbuild` from the `build.cmd` script in the project. The `dotnet build` seems to do the same as `dotnet msbuild`, but just to be sure we are not committing something invalid, do you know if there's any difference for the project between the two ? (other than `dotnet build` implicitly running `dotnet restore`)

Reply:
as far as I understand it's very similar except for the way you pass arguments to msbuild. dotnet msbuild being closer to the msbuild syntax from back in the days. https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-msbuild#description